app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -0,0 +1,128 @@
+class ManageIQ::Providers::Vmware::CloudManager::RefreshParser < ManageIQ::Providers::CloudManager::RefreshParser
+  def self.ems_inv_to_hashes(ems, options = nil)
+    new(ems, options).ems_inv_to_hashes
+  end
+
+  def initialize(ems, options = Config::Options.new)
+    @ems                 = ems
+    @data                = {}
+    @data_index          = {}
+    @options             = options
+  end
+
+  def process_collection(collection, key)
+    @data[key] ||= []
+
+    collection.each do |item|
+      uid, new_result = yield(item)
+      next if uid.nil?
+
+      @data[key] << new_result
+      @data_index.store_path(key, uid, new_result)
+    end
+  end
+
+  def ems_inv_to_hashes
+    log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{@ems.name}] id: [#{@ems.id}]"
+
+    $log.info("#{log_header}...")
+
+    vcd = @ems.vcd_service
+
+    @ems.api_version = vcd.api_version
+
+    @inv = {
+      :orgs  => [],
+      :vdcs  => [],
+      :vapps => [],
+      :vms   => [],
+    }
```

Comment:
Should probably set this in `#initialize`, you can do something like `@inv = Hash.new { |h, k| h[k] = [] }` so you don't have to initialize each key individually